### PR TITLE
Week1

### PR DESCRIPTION
**Pull request by chichiglacierz**

### CODE REVIEW

File: homework/index.js [hunk 1]
```diff
@@ -33,14 +36,66 @@
...
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        //createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const header = createAndAppend('header', root, { class: 'header' });
+        const h1 = createAndAppend('h1', header, { text: 'FooCoding Repos', class: 'h1' });
```
Comment (mattiaslundberg) [inline homework/index.js:41]: Nice usage of `createAndAppend`!

File: homework/index.js [hunk 2]
```diff
@@ -33,14 +36,66 @@
...
+
+        const select = createAndAppend('select', root, { class: 'select' });
+        createAndAppend('option', select, { text: 'Choose your favorite repo below:' });
```
Comment (mattiaslundberg) [inline homework/index.js:44]: One of the requirements is that information for the first repo is shown when the page loads, can you have a look at that?

File: homework/index.js [hunk 3]
```diff
@@ -33,14 +36,66 @@
...
+
+        data.forEach(repo => {
+          const name = repo.name;
+          createAndAppend('option', select, { text: name });
+          //console.log(name);
```
Comment (mattiaslundberg) [inline homework/index.js]: Don't leave commented out code, remove it! It's easier for me to review and easier for you to read when you come back to it at a later time.

File: homework/index.js [hunk 4]
```diff
@@ -33,14 +36,66 @@
...
+          const addInfo = (label, value) => {
+            const container = createAndAppend('div', repoInfo);
+            createAndAppend('span', container, { text: label });
+            createAndAppend('span', container, { text: value });
+          };
+          addInfo('Name: ', repo.name);
+          addInfo('Description: ', repo.description);
+          addInfo('Number of forks: ', repo.forks);
+          addInfo('Updated: ', repo.updated_at);
+          //or instead of using the addInfo function we can use these lines:
```
Comment (mattiaslundberg) [inline homework/index.js]: To create a link:
`createAndAppend('a', repoInfo, {text: label, href: url})`

File: homework/index.js [hunk 5]
```diff
@@ -33,14 +36,66 @@
...
+          // createAndAppend('div', repoInfo, { text: `Updated at:${repo.updated_at}` });
+          const contribsUrl = repo.contributors_url;
+          fetchJSON(contribsUrl, (err, contribData) => {
+            if (err) {
+              createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+            } else {
+            }
+            contribData.forEach(contributor => {
+              createAndAppend('div', contribs, { text: contributor.login });
+              //createAndAppend('a', contribs, { href: contributor.html.url, setAttribute: 'href' });
```
Comment (mattiaslundberg) [inline homework/index.js]: I think the issues here are:

 1. `contributor.html.url` should be `contributor.html_url`
 2. you need to add `text: contributor.login`

File: homework/index.js [hunk 6]
```diff
@@ -33,14 +36,66 @@
...
+              createAndAppend('img', contribs, {
```
Comment (mattiaslundberg) [inline homework/index.js:86]: Nice!

**Review by mattiaslundberg — CHANGES_REQUESTED** [review]
Missing two things (other comments are nice to have/fix later but optional):

 1. Create the link (there are some pointers in the comment)
 2. Load repo information when the page loads (let me know if you need more pointers for this!)

Almost

File: homework/index.js (hunk 5 above)
Comment (chichiglacierz) [reply homework/index.js]: So that's why it wasn't working. Thanks!

File: homework/index.js (hunk 2 above)
Comment (chichiglacierz) [reply homework/index.js:44]: Trying...

**Pull request closed without merging**